_aggregations/pipeline/index.md

```diff
@@ -0,0 +1,214 @@
+  }
+}
+```
+{% include copy-curl.html %}
+
+Note that the `buckets_path` contains the names of the component aggregations. Paths are directed, meaning that they cascade one way, downward from parents to children. 
+
+### Buckets path example response
+
+The pipeline aggregation returns the total memory summed from all the buckets:
+
+```json
+{
+  ...
+  "aggregations": {
+    "number_of_bytes": {
+      "buckets": [
+        {
+          "key": 0,
+          "doc_count": 13372,
+          "sum_total_memory": {
+            "value": 91266400
+          }
+        },
+        {
+          "key": 10000,
+          "doc_count": 702,
+          "sum_total_memory": {
+            "value": 0
+          }
+        }
+      ]
+    },
+    "sum_copies": {
+      "value": 91266400
+    }
+  }
+}
+```
+
+### Count paths
+
+You can direct the `buckets_path` to use a count rather than a value as its input. To do so, use the special `_count` buckets path variable.
+
+The following example computes basic stats on a histogram of the number of bytes from the OpenSearch Dashboards logs sample data. It creates a histogram of values in the `bytes` field, then computes the stats on the counts in the histogram buckets.
+
+```json
+GET opensearch_dashboards_sample_data_logs/_search
+{
+  "size": 0,
+  "aggs": {
+    "number_of_bytes": {
+      "histogram": {
+        "field": "bytes",
+        "interval": 10000
+      }
+    },
+    "count_stats": {
+      "stats_bucket": {
+        "buckets_path": "number_of_bytes>_count"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The results show stats on the *document counts* of the buckets:
+
+```json
+{
+...
+  "aggregations": {
+    "number_of_bytes": {
+      "buckets": [
+        {
+          "key": 0,
+          "doc_count": 13372
+        },
+        {
+          "key": 10000,
+          "doc_count": 702
+        }
+      ]
+    },
+    "count_stats": {
+      "count": 2,
+      "min": 702,
+      "max": 13372,
+      "avg": 7037,
+      "sum": 14074
+    }
+  }
+}
+```
+
+## Data gaps
+
+Data can be missing from nested aggregations for a number of reasons, including:
+
+- Missing values in documents
+- Empty buckets anywhere in the chain of aggregations
+- Data needed to calculate a bucket value is missing (for example, rolling functions such as `derivative` require one or more previous values to start).
+
+You can specify a policy to deal with missing data using the `gap_policy` property. 
+
+You can set the `gap_policy` property to do one of three things:
+
+- Skip the missing data
+- Use non-numerical data if it is non-null and not `NaN`, otherwise skip the data
+- Replace the missing data with zeros
+
+The `gap_policy` parameter is valid with all pipeline aggregations. The valid values for the parameter are:
+
+| Value | Action |
+| -- | -- |
+| `skip` | Ignore the bucket containing the missing data. |
+|`keep_values` | Skip buckets with `null` and `NaN` values; otherwise use the provided value. |
```

Comment:
While `skip` and `insert_zeros` are pretty clear (either the bucket is dropped or it's inserted with a value of `0`), I don't understand how `keep_values` works. Maybe a short example showing all three would help?

Reply:
@msfroh I'm not sure either, since there wasn't an example in the existing page. The key question is, What other values besides null and NaN would cause the system to omit a document? A non-numerical format? But then how would it aggregate that with numerical data? Might have to look in the code to find out.

Reply:
```suggestion
```